modules/arrow/src/geoarrow/convert-geojson-feature-to-arrow.ts

```diff
@@ -0,0 +1,319 @@
+  const field = new ArrowField(name, pointType, nullable, metaData);
+
+  // make geoarrow builder
+  const builder = makeBuilder({
+    type: pointType,
+    nullValues: [null]
+  });
+
+  // fill builder with coordinates
+  for (let i = 0; i < points.length; i++) {
+    const coords = (points[i].geometry as Point).coordinates;
+    // @ts-ignore
+    builder.append(coords);
+  }
+
+  // build geometry vector
+  const geometry = builder.finish().toVector();
+
+  return {
+    field,
+    geometry
+  };
+}
+
+/**
+ * convert GeoJSON Point to arrow.Vector
+ */
+export function geojsonMultiPointToArrow(name: string, points: Feature[]): GeoArrowReturnType {
+  // get dimension from the first multipoint
+  const dimension = (points[0].geometry as MultiPoint).coordinates[0].length;
+  const pointFieldName = dimension === 2 ? 'xy' : 'xyz';
+
+  // get multipoint type
+  const nullable = false;
+  const coordType = new ArrowFloat();
+  const pointType = new FixedSizeList(
+    dimension,
+    new ArrowField(pointFieldName, coordType, nullable)
+  );
+
+  const multiPointType = new ArrowList(new ArrowField('points', pointType, nullable));
+
+  // create a field
+  const metaData: Map<string, string> = new Map([['ARROW:extension:name', 'geoarrow.multipoint']]);
+  const field = new ArrowField(name, multiPointType, nullable, metaData);
+
+  // make geoarrow builder
+  const builder = makeBuilder({
+    type: multiPointType,
+    nullValues: [null]
+  });
+
+  // const pointBuilder = builder.getChildAt(0);
+  for (let i = 0; i < points.length; i++) {
+    const coords = (points[i].geometry as MultiPoint).coordinates;
+    // @ts-ignore
+    builder.append(coords);
+  }
+
+  const geometry = builder.finish().toVector();
+
+  return {
+    field,
+    geometry
+  };
+}
+
+/**
+ * Convert GeoJSON LineString to arrow.Data
+ */
+export function geojsonLineStringToArrow(name: string, lines: Feature[]): GeoArrowReturnType {
+  // get dimension from the first line
+  const dimension = (lines[0].geometry as LineString).coordinates[0].length;
+  const pointFieldName = dimension === 2 ? 'xy' : 'xyz';
+
+  // get line type
+  const nullable = false;
+  const coordType = new ArrowFloat();
+  const pointType = new FixedSizeList(
+    dimension,
+    new ArrowField(pointFieldName, coordType, nullable)
+  );
+  const lineType = new ArrowList(new ArrowField('points', pointType, nullable));
+
+  // create a field
+  const metaData: Map<string, string> = new Map([['ARROW:extension:name', 'geoarrow.linestring']]);
+  const field = new ArrowField(name, lineType, nullable, metaData);
+
+  // make geoarrow builder
+  const builder = makeBuilder({
+    type: lineType,
+    nullValues: [null]
+  });
+
+  for (let i = 0; i < lines.length; i++) {
+    const coords = (lines[i].geometry as LineString).coordinates;
+    // @ts-ignore
+    builder.append(coords);
+  }
+
+  const geometry = builder.finish().toVector();
+
+  return {
+    field,
+    geometry
+  };
+}
```

Comment:
It seems like each of these 6 functions are exactly the same except for the creation of the types. You might be able to deduplicate the second half (filling) of each of these functions.